azurecompute/src/main/java/org/jclouds/azurecompute/compute/AzureComputeServiceAdapter.java

```diff
@@ -16,39 +16,275 @@
+   private static final String DEFAULT_LOGIN_PASSWORD = "Azur3Compute!";
+   private static final String DEFAULT_STORAGE_ACCOUNT_PREFIX = "jclouds";
+
+   @Resource
+   @Named(ComputeServiceConstants.COMPUTE_LOGGER)
+   protected Logger logger = Logger.NULL;
+
+   private final AzureComputeApi api;
+   private final String subscriptionId;
+   private final long operationTimeout;
+   private Predicate<String> operationSucceeded;
+
+   @Inject
+   public AzureComputeServiceAdapter(final AzureComputeApi api, @Named(SUBSCRIPTION_ID) String subscriptionId, @Named(OPERATION_TIMEOUT) long operationTimeout) {
+      this.api = checkNotNull(api, "api");
+      this.subscriptionId = checkNotNull(subscriptionId, "subscriptionId");
+      this.operationTimeout = checkNotNull(operationTimeout, "operationTimeout");
+      this.operationSucceeded = retry(new Predicate<String>() {
+         public boolean apply(String input) {
+            return api.getOperationApi().get(input).status() == Operation.Status.SUCCEEDED;
+         }
+      }, operationTimeout, 5, 5, SECONDS);
+   }
 
    @Override
    public NodeAndInitialCredentials<Deployment> createNodeWithGroupEncodedIntoName(
-         String group, String name, Template template) {
-      // TODO Auto-generated method stub
-      return null;
+           final String group, final String name, Template template) {
+
+      // azure-specific options
+      AzureComputeTemplateOptions templateOptions = template.getOptions().as(AzureComputeTemplateOptions.class);
+      final String virtualNetworkName = templateOptions.getVirtualNetworkName().or(DEFAULT_VIRTUAL_NETWORK_NAME);
+      final String addressSpaceAddressPrefix = templateOptions.getAddressSpaceAddressPrefix().or(DEFAULT_ADDRESS_SPACE_ADDRESS_PREFIX);
+      final String subnetName = templateOptions.getSubnetName().or(DEFAULT_SUBNET_NAME);
+      final Set<String> networkSecurityGroupNames = templateOptions.getGroups().isEmpty() ? Sets.<String>newHashSet() : templateOptions.getGroups();
+      final String subnetAddressPrefix = templateOptions.getSubnetAddressPrefix().or(DEFAULT_SUBNET_ADDRESS_PREFIX);
+      final String storageAccountType = templateOptions.getStorageAccountType().or(DEFAULT_STORAGE_SERVICE_TYPE);
+
+      final String loginUser = templateOptions.getLoginUser() != null ? templateOptions.getLoginUser() : DEFAULT_LOGIN_USER;
+      final String loginPassword = templateOptions.getLoginPassword() != null ? templateOptions.getLoginPassword() : DEFAULT_LOGIN_PASSWORD;
+      final String location = template.getLocation().getId();
+      final int[] inboundPorts = template.getOptions().getInboundPorts();
+
+      String storageAccountName = templateOptions.getStorageAccountName().or(generateStorageServiceName(DEFAULT_STORAGE_ACCOUNT_PREFIX));
+
+      // get or create storage service
+      StorageService storageService = tryFindExistingStorageServiceAccountOrCreate(location, storageAccountName, storageAccountType);
```

Comment:
Take care with `get or create` operations here, as they will introduce race conditions. This method will be executed concurrently when creating more than one node.
This should be done in the upper layer, in a custom implementation of the `CreateNodesWithGroupEncodedIntoNameThenAddToSet` class. Take a look at how [digitalocean creates the key pairs globally](https://github.com/jclouds/jclouds-labs/blob/master/digitalocean/src/main/java/org/jclouds/digitalocean/compute/strategy/CreateKeyPairsThenCreateNodes.java) before creating the nodes to avoid this kind of race conditions.